crates/primitives/transactions/src/execution.rs

```diff
@@ -83,6 +86,47 @@ impl SimulateTxVersionOffset for TransactionVersion {
     }
 }
 
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Encode, Decode)]
+#[cfg_attr(feature = "scale-info", derive(scale_info::TypeInfo))]
+#[cfg_attr(feature = "no_unknown_fields", serde(deny_unknown_fields))]
+pub struct StarknetRPCExecutionResources {
+    /// The number of cairo steps used
+    pub steps: u64,
+    /// The number of unused memory cells (each cell is roughly equivalent to a step)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub memory_holes: Option<u64>,
+    /// The number of range_check builtin instances
+    pub range_check_builtin_applications: u64,
+    /// The number of pedersen builtin instances
+    pub pedersen_builtin_applications: u64,
+    /// The number of poseidon builtin instances
+    pub poseidon_builtin_applications: u64,
+    /// The number of ec_op builtin instances
+    pub ec_op_builtin_applications: u64,
+    /// The number of ecdsa builtin instances
+    pub ecdsa_builtin_applications: u64,
+    /// The number of bitwise builtin instances
+    pub bitwise_builtin_applications: u64,
+    /// The number of keccak builtin instances
+    pub keccak_builtin_applications: u64,
+}
+
+impl StarknetRPCExecutionResources {
+    pub fn into_core_execution_resources(self) -> CoreExecutionResources {
```

Comment:
Why not use `From` trait?

Reply:
It was handled by an external contributor, I corrected it.